pkg/downwardmetrics/virtio-serial/server.go

```diff
@@ -0,0 +1,279 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package virtio_serial
+
+import (
+	"bufio"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/textproto"
+	"net/url"
+	"strings"
+	"syscall"
+	"time"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/downwardmetrics"
+	"kubevirt.io/kubevirt/pkg/downwardmetrics/vhostmd/api"
+	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+	metricsScraper "kubevirt.io/kubevirt/pkg/monitoring/domainstats/downwardmetrics"
+	"kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+const (
+	maxConnectAttempts         = 6
+	standardLauncherSocketPath = "/run/kubevirt/sockets/launcher-sock"
+	invalidRequest             = "INVALID REQUEST\n\n"
+	emptyMetrics               = "<metrics><!-- host metrics not available --><!-- VM metrics not available --></metrics>"
+)
+
+func RunDownwardMetricsVirtioServer(nodeName string, stop chan struct{}) error {
+	report, err := newMetricsReporter(nodeName)
+	if err != nil {
+		return err
+	}
+
+	server := downwardMetricsServer{
+		maxConnectAttempts: maxConnectAttempts,
+		virtioSerialSocket: downwardmetrics.DownwardMetricsChannelSocket,
+		reportFn:           report,
+	}
+	return server.start(stop)
+}
+
+type metricsReporter func() (*api.Metrics, error)
+
+func newMetricsReporter(nodeName string) (metricsReporter, error) {
+	exists, err := diskutils.FileExists(standardLauncherSocketPath)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.New("virt-launcher socket not found")
+	}
+
+	isolationDetector := isolation.NewSocketBasedIsolationDetector(util.VirtShareDir)
```

Comment:
isolation detector is normally used by the virt-handler in order to find the virt-launcher's pod compute container. virt-handler then can chroot into the container and perform operations in the container.
I'm not sure why do we need this in this case as we are already in the virt-launcher's compute container....

Reply:
Yes I know, we don't really use this (it's used inside of Scrape() that we don't call), it's just a requirement to create the Scraper object. I can think in 3 options:
1- Just leave a comment explaining it
2- Replace `isolationDetector` with a `nil` (i.e., calling  `scraper := metricsScraper.NewScraper(nodeName, nil)`)  (shivers :) )
3- create a `EmptyIsolationDetector` the implements `isolation.PodIsolationDetector` always returning error

WDYT?